ccdproc/bitmask.py

```diff
@@ -0,0 +1,473 @@
+            The number of "layers" of the boolean array. This parameter is
+            mainly so that the function doesn't have to guess how many
+            different flags the bitmask could have and to minimize the result
+            of the final boolean array. If ``None`` the number of flags is
+            determined by the ``dtype`` of the ``bitmask``.
+            Default is ``None``.
+
+        Returns
+        -------
+        boolarray : `numpy.ndarray` of boolean dtype
+            The boolean representation of the bitmask. It will have one more
+            dimension that the original ``bitmask`` where the first dimension
+            represents the "flags". So ``boolarray[0]`` will return an array of
+            the size of the original ``bitmask`` and with values ``False`` (if
+            the first flag wasn't set) and ``True`` (if the first flag was
+            set).
+
+        Examples
+        --------
+        This converts the bitmask to its boolean representation. For example
+        if it's easier to manipulate::
+
+            >>> from ccdproc import Bitmask
+            >>> bm = Bitmask(np.arange(7))
+            >>> boolbitmask = bm.to_boolarray(num=3)
+            >>> boolbitmask
+            array([[False,  True, False,  True, False,  True, False],
+                   [False, False,  True,  True, False, False,  True],
+                   [False, False, False, False,  True,  True,  True]], dtype=bool)
+
+        The first index represents a flag, so by accessing the booleans by
+        single integers allows to manipulate, reset, overwrite a given flag
+        for the bitmask easily::
+
+            >>> boolbitmask[1]  # Get the values for the second flag
+            array([False, False,  True,  True, False, False,  True], dtype=bool)
+
+            >>> boolbitmask[0] = 0  # set all values for the first flag to 0
+
+        Similarly the boolean array can be converted to a bitmask again::
+
+            >>> Bitmask.from_boolarray(boolbitmask)
+            Bitmask (
+            array([0, 0, 2, 2, 4, 4, 6], dtype=int8)
+            name_map={}
+            )
+
+        Notes
+        -----
+        The conversion from and to boolean representation can be quite slow so
+        shouldn't be used excessivly.
+        """
+        bitmask = self.data
+        if num is not None and 0 > num > len(_POWERS) - 1:
+            raise ValueError(
+                '"num" must be between 0 and {} (both inclusive).'
+                .format(len(_POWERS) - 1))
+        # num is a space-optimization, if there is no "num" given
+        # get the number of representable flags by the dtype of
+        # the bitmask. An alternative would be to find the maximum
+        # and use the maximal power of 2 that is smaller or equal
+        # to the maximum. However hat would require a whole pass
+        # over the array and using the dtype makes sure we use
+        # exactly the same memory as the original array!
```

Comment:
ups, I forget the word "amount". That should mean "it uses the same amount of memory". Not the same memory on disk